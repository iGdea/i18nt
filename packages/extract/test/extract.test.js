const { describe, it, expect } = require('@jest/globals');
const { extract } = require('../');

describe('#extract', () => {
  it('#ts', () => {
    const result = extract({
      fileName: '/root/example.ts',
      fileContent: `
      i18n('i18n text');
      i18n('i18n text', 'subkey1');
      `,
    });

    expect(result).toMatchSnapshot();
  });


  it('#vue', () => {
    const result = extract({
      fileName: '/root/example.vue',
      fileContent: `
      <template>
        <div :xxx1="i18n('i18n attr')">
          {{i18n('i18n text')}}
        </div>
      </template>
      <script>
        i18n('i18n script');
        i18n('i18n script', 'subkey1');
      </script>
      `,
    });

    expect(result).toMatchSnapshot();
  });
});
