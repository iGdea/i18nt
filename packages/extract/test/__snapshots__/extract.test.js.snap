// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#extract #ts 1`] = `
{
  "list": [
    {
      "subkey": undefined,
      "text": "i18n text",
    },
    {
      "subkey": "subkey1",
      "text": "i18n text",
    },
  ],
}
`;

exports[`#extract #vue 1`] = `
{
  "list": [
    {
      "subkey": undefined,
      "text": "i18n text",
    },
    {
      "subkey": undefined,
      "text": "i18n attr",
    },
    {
      "subkey": undefined,
      "text": "i18n script",
    },
    {
      "subkey": "subkey1",
      "text": "i18n script",
    },
  ],
}
`;
